### ISSUE TITLE
Introduce wildcards support for aws_s3 input 

### ISSUE THREAD

**Issue by bigliolimatteo** [Jeffail/benthos#1148]
Hey there 😃 

**TL;DR:** It would be great to introduce wildcards support for the aws_s3 input.

**LONG:** 
I'm trying to read from an `aws_s3` inputs a bunch of csv in a bucket+prefix with a csv codec (`codec: "csv:|"`)

I noticed that I am able to do that only if in the specified prefix there are only `*.csv` files

The problem is that, because I am generating those files via spark, I often get also a `_SUCCESS` file or other meta stuff 

![image](https://user-images.githubusercontent.com/46928368/155496922-ccdf9092-8dc6-473f-876d-d418bb80f664.png)

With wildcards we could easily specify a prefix like `bucket/prefix/*.csv`

**Comment by bigliolimatteo** [comment]
For anyone looking at this issue, a temporary fix suggested by @Jeffail is the following:

>  If the files are small enough to process them in-memory then use `all-bytes` and then add a processor that removes the messages from `_SUCCESS` files before parsing it as CSV, something like this should work:
```
pipeline:
  processors:
    - bloblang: |
        root = if meta("s3_key").has_suffix("_SUCCESS") {
          deleted()
        } else {
          content().parse_csv()
        }
```

**Comment by twmb** [comment]
The S3 input supports prefix filtering, and additional filtering can be done with bloblang processors as shown in the comments. Closing as stale. If native wildcard/glob support is still needed, feel free to open a new issue or PR.
